7.22 add tomcat 10 documentation (#1647)
camunda/camunda-docs-manual: content/update/minor/721-to-722/_index.md

1. For developers: [Camunda Template Engines FreeMarker](#camunda-template-engines-freemarker)
1. For developers: [Camunda Connect](#camunda-connect)
1. For administrators and developers: [Update to JBoss EAP 8.0](#update-to-jboss-eap-8)

> 1. For developers: [Update to Tomcat 10 Server](#update-to-tomcat-10-server)

Reviewer: 🔧 Also important for admins.

```suggestion
1. For administrators and developers: [Update to Tomcat 10 Server](#update-to-tomcat-10-server)
```

# Camunda Connect
 We’ve moved the `camunda-connect` project from its [previous location](https://github.com/camunda/camunda-connect) into the [mono repository](https://github.com/camunda/camunda-bpm-platform). We’re no longer versioning it independently. Instead, we’ve integrated it into the 7.X.Y versioning scheme, so you can conveniently declare Camunda `7.22.0-alpha2` to use the latest release of Camunda Connect.

# Update to Tomcat 10 Server

> With this release, we support Tomcat 10, the new default pre-packaged Tomcat distro. It replaces Java EE with Jakarta EE APIs.

Reviewer: > With this release, we support Tomcat 10, the new default pre-packaged Tomcat distro.

I'd split these infos.
1. We have now support for Tomcat 10 which is great because... It supports this and that.
2. From now on, our pre-packaged tomcat distro is built with Tomcat 10.
    - I'd also note that our docker image will also use Tomcat 10 from now on.

Author: I've rewrote and split the informative statement for tomcat 10 into two sections according to your suggestion ([commit](https://github.com/camunda/camunda-docs-manual/pull/1647/commits/87480758c0a79c2154071eb9a31fce288d842d0e))

I think it's indeed more clear and separates the two important announcements:
a) emphasis on tomcat 10 reasons to ugprade
b) the new version will be based on tomcat 10

If you prefer to stay on Tomcat 9, you can still download the Java EE compliant [web application][tomcat9-webapp], and [REST API][tomcat9-rest-api].

To work with Tomcat 10, consider the following when migrating your process applications and replacing artifacts on the application server:

[tomcat9-webapp]: https://artifacts.camunda.com/ui/native/camunda-bpm/org/camunda/bpm/webapp/camunda-webapp-tomcat/
[tomcat9-rest-api]: https://artifacts.camunda.com/artifactory/public/org/camunda/bpm/camunda-engine-rest/

> ### Migrate process applications

Reviewer: 👍 Nice migration guides!

### Replace REST API deployment

Replace the artifact `camunda-engine-rest-$PLATFORM_VERSION-wildfly.war` with `camunda-engine-rest-jakarta-$PLATFORM_VERSION-wildfly.war` under `$JBOSS_HOME/standalone/deployments`.

# Update to Tomcat 10 Server

This version brings support for `Tomcat 10.1`. A few reasons to upgrade are:

* Namespace Change: Switch from `javax.*` to `jakarta.*` for future compatibility.
* Enhanced Security: Improved security features and fixes.
* Modern Features: Supports `Servlet 6.0`, `JSP 3.1`, and `WebSocket 2.1`.
* Performance Improvements: Faster response times and better resource efficiency.
* Simplified Migration: Tools and documentation for easier transition from earlier versions.
* Better Integration: Enhanced compatibility with `Jakarta EE` components and third-party libraries.

From now on, our pre-packaged Tomcat distribution is built with `Tomcat 10.1`.

> Additionally, the Docker image distribution, once configured to use Tomcat, will also utilize `Tomcat 10.1`.

Reviewer: > once configured to use Tomcat

Ah okay, it's a separate ticket which will be done at a later point. I forgot about that.
🔧 I think you could still write it as if it was already done. It will be finished soon I guess and until the 7.22 minor. Or you can update this later during the docker ticket.

Author: I wrote it like that to reflect that the docker image has different [configuration](https://github.com/camunda/docker-camunda-bpm-platform#:~:text=Apache%20Tomcat%20%2D,Platform%20Run%20%2D%20documentation) and if the user selects Tomcat, then Tomcat 10 will be used.

We don't have to wait for the docker task, if you find the phrasing unclear, we can remove the "once configured to use Tomcat" part.

Reviewer: I misunderstood it then. I feel like it's a nitpick so I leave this up to you. 🙃

Author: Let's keep it as-is for now then. If that part sounds confusing anybody else, i'll revisit this point and remove that sentence

> Additionally, the Docker image distribution, once configured to use Tomcat, will also utilize `Tomcat 10.1`.

Reviewer 2: ```suggestion
Additionally, the Tomcat Docker image will, from now on, utilize `Tomcat 10.1`.
```

🔧 I was confused like Daniel was. This migration guide section is about Tomcat. Why do we need to explicitly mention this applies only when configured? Also, there is no real configuration option. The user just pulls the image with the tomcat tag from our registry.

### Migrate process applications

* Replace Java EE class references (`javax.*`) with Jakarta class references (`jakarta.*`)
 * You might have a look at [`org.eclipse.transformer:transformer-maven-plugin`](https://github.com/eclipse/transformer)
* Replace Camunda class references:
 * `org.camunda.bpm.application.impl.EjbProcessApplication` → `org.camunda.bpm.application.impl.JakartaEjbProcessApplication`
 * `org.camunda.bpm.application.impl.ServletProcessApplicationDeployer` → `org.camunda.bpm.application.impl.JakartaServletProcessApplicationDeployer`
 * `org.camunda.bpm.application.impl.ServletProcessApplication` → `org.camunda.bpm.application.impl.JakartaServletProcessApplication`
 * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContext` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContext`
 * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContextFactory` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContextFactory`
 * `org.camunda.bpm.engine.impl.cfg.JtaProcessEngineConfiguration` → `org.camunda.bpm.engine.impl.cfg.JakartaTransactionProcessEngineConfiguration`

>  * `org.camunda.bpm.engine.impl.interceptor.JtaTransactionInterceptor` → `org.camunda.bpm.engine.impl.interceptor.JakartaTransactionInterceptor`

Reviewer 2: ❓ Are we sure all these classes are relevant in the scope of Tomcat? Do we support scenarios in Tomcat for all of these classes?

Author: | Class Name | Utility Description|
|-------------|------------------|
|`JakartaEjbProcessApplication`| Can be used when users utilise `Tomcat 10` and `EJBs` |
|`JakartaServletProcessApplicationDeployer` | Jakarta version of a class that will add process applications declared in the `META/INF/processes.xml` folder as listeners |
|`JakartaServletProcessApplication` | Related to the above |
|`JakartaTransactionContext`  | Although Tomcat does not support JTA transactions out of the box like Application |Servers do, a user could add a transaction manager of their choice, in which case this class looks relevant. |
|`JakartaTransactionContextFactory` | Same as above |
|`JakartaTransactionProcessEngineConfiguration` | Same as above |
|`JakartaTransactionInterceptor` | Same as above |

Reviewer: I didn't do a full research but I think EJB is not necessary as Tomcat is only a servlet container (no EE features) and JTA classes are probably not applicable either.

Author: - In theory, a user could use Tomcat + OpenEJB instead of [Tomee](https://tomee.apache.org/download.html)to have EJB support.
- Similarly, can't a user add a Transaction Manager to Tomcat like [Atomikos](https://www.atomikos.com/Main/WebHome) and have JTA support?

Reviewer 2: Might be possible. I'm fine with just leaving this wording as it is. It doesn't harm. 👍

* Replace Camunda Maven dependencies:
 * `org.camunda.bpm.javaee:camunda-ejb-client` → `org.camunda.bpm.javaee:camunda-ejb-client-jakarta`
 * `org.camunda.bpm:camunda-engine-cdi` → `org.camunda.bpm:camunda-engine-cdi-jakarta`

### Migrate Java webapp plugins

Replace Java EE class references (`javax.*`) with Jakarta class references (`jakarta.*`)

### Replace web application (Cockpit, Admin, Tasklist, Welcome) deployment

Replace the artifact `camunda-webapp-tomcat-$PLATFORM_VERSION.war` with `camunda-webapp-tomcat-jakarta-$PLATFORM_VERSION.war` under `$CATALINA_HOME/webapps`.

### Replace REST API deployment

Replace the artifact `camunda-engine-rest-$PLATFORM_VERSION-tomcat.war` with `camunda-engine-rest-jakarta-$PLATFORM_VERSION-tomcat.war` under `$CATALINA_HOME/webapps`.

### Migrating to the Tomcat 10 Docker Image

> If your application uses a Docker image based on `Tomcat 9` (or prior) and wish to use the `Tomcat 10` based **Docker image**, you need perform the

Author: Adjusted the [message](https://github.com/camunda/camunda-docs-manual/pull/1647/commits/67b805e5a5ed3760990b42141f35b1ec1b23073e) according to the suggestion
